File: AbetOrder/OpenDataAccess/OpenDataAccess.Entity/IEntityData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OpenDataAccess.Entity
{
    public interface IEntityData
    {
        string EntityName { get; set; }
        object GetValue(string propertyName);
        object GetValue(int i);
        void SetValue(string propertyName, object value);
        void SetValue(int i, object value);
        void Remove(string propertyName);
        void Remove(int i);
        IEntity ToEntity<T>() where T : IEntity;
        Dictionary<string, object> ToDictionary();
        Dictionary<string, object> GetDictionary();
        int Count { get; }
        string GetPropertyName(int i);
        string GetStringValue(string propertyName);
        T GetValue<T>(string propertyName);
    }
}

File: JavaOpenDataAccess/src/OpenDataAccess/Entity/IEntityData.java
package OpenDataAccess.Entity;

import java.util.Map;

public interface IEntityData {
    String EntityName(String value, boolean blGet);

    Object GetValue(String propertyName);

    Object GetValue(int i);

    void SetValue(String propertyName, Object value);

    void SetValue(int i, Object value);

    void Remove(String propertyName);

    void Remove(int i);

    <T extends IEntity> IEntity ToEntity(Class<T> cls) throws IllegalAccessException, InstantiationException ;

    Map<String, Object> ToDictionary();

    Map<String, Object> GetDictionary();

    int Count();

    String GetPropertyName(int i);

    String GetStringValue(String propertyName);

    <T> T GetValueByType(Class<T> cls, String propertyName) throws Exception;

    void SetDefaultValue(String property, Object value);
}
